wormhole_chain/app/cosmwasm_handlers.go

```diff
@@ -0,0 +1,213 @@
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ibcappkeeper "github.com/cosmos/ibc-go/v3/modules/apps/transfer/keeper"
+	connectiontypes "github.com/cosmos/ibc-go/v3/modules/core/03-connection/types"
+	ibckeeper "github.com/cosmos/ibc-go/v3/modules/core/04-channel/keeper"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	ibcportkeeper "github.com/cosmos/ibc-go/v3/modules/core/05-port/keeper"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+)
+
+// This defines which modules we actually want to expose cosmwasm contracts.
+// Right now we only permit methods that are read-only.
+// See https://github.com/CosmWasm/wasmd/blob/d63bea442bedf5b3055f3821472c7e6cafc3d813/x/wasm/types/expected_keepers.go
+
+type BankViewKeeperHandler struct {
+	Keeper bankkeeper.Keeper
+}
+type BurnerHandler struct {
+	Keeper bankkeeper.Keeper
+}
+
+type BankKeeperHandler struct {
+	BankViewKeeperHandler
+	BurnerHandler
+}
+
+type AccountKeeperHandler struct {
+	AccountKeeper authkeeper.AccountKeeper
+}
+type DistributionKeeperHandler struct {
+	Keeper distrkeeper.Keeper
+}
+type StakingKeeperHandler struct {
+	Keeper stakingkeeper.Keeper
+}
+type ChannelKeeperHandler struct {
+	Keeper ibckeeper.Keeper
+}
+type ClientKeeperHandler struct {
+	Keeper ibckeeper.Keeper
+}
+type ConnectionKeeperHandler struct {
+	Keeper ibckeeper.Keeper
+}
+type PortKeeperHandler struct {
+	Keeper ibcportkeeper.Keeper
+}
+type CapabilityKeeperHandler struct {
+	ScopedKeeper capabilitykeeper.ScopedKeeper
+}
+type ICS20TransferPortSourceHandler struct {
+	Keeper ibcappkeeper.Keeper
+}
+
+var _ wasmtypes.BankViewKeeper = &BankViewKeeperHandler{}
+
+func (b *BankViewKeeperHandler) GetAllBalances(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
+	return b.Keeper.GetAllBalances(ctx, addr)
+}
+func (b *BankViewKeeperHandler) GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	return b.Keeper.GetBalance(ctx, addr, denom)
+}
+
+var _ wasmtypes.Burner = &BurnerHandler{}
+
+func (b *BurnerHandler) BurnCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) error {
+	return errors.New("not permitted")
+}
+
+func (b *BurnerHandler) SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error {
+	return errors.New("not permitted")
+}
+
+var _ wasmtypes.BankKeeper = &BankKeeperHandler{}
+
+func (b *BankKeeperHandler) IsSendEnabledCoins(ctx sdk.Context, coins ...sdk.Coin) error {
+	return errors.New("not permitted")
+}
+func (b *BankKeeperHandler) BlockedAddr(addr sdk.AccAddress) bool {
+	return false
+}
+func (b *BankKeeperHandler) SendCoins(ctx sdk.Context, fromAddr sdk.AccAddress, toAddr sdk.AccAddress, amt sdk.Coins) error {
+	return errors.New("not permitted")
+}
```

Comment:
I think we will end up replacing the current native tokenbridge module with the tokenbridge smart contract that we use on all other cosmwasm chains.  At that point, we'll likely need to start allowing these methods.

Reply:
Added